### PR TITLE
issue #276627. fixed signatures for winmm midi callback functions 

### PR DESCRIPTION
**Pull request by L0gi**
should now compile and run correctly with msvc on Win_x64

### CODE REVIEW

**Comment by Jojo-Schmitz** [comment]
Patch taken from https://www.mail-archive.com/media_api@create.ucsb.edu/msg00513.html

File: thirdparty/portmidi/pm_win/pmwinmm.c [hunk 1]
```diff
@@ -1,12 +1,13 @@
 /* pmwinmm.c -- system specific definitions */
-
```
Comment (Jojo-Schmitz) [inline thirdparty/portmidi/pm_win/pmwinmm.c]: Please avoid all pure whitespace changes, in order to keep the patch as small as possible

File: thirdparty/portmidi/pm_win/pmwinmm.c [hunk 2]
```diff
@@ -1,12 +1,13 @@
...
 #ifndef _WIN32_WINNT
     /* without this define, InitializeCriticalSectionAndSpinCount is 
      * undefined. This version level means "Windows 2000 and higher" 
      */
     #define _WIN32_WINNT 0x0500
+
 #endif
 
+#include <stdint.h>
```
Comment (Jojo-Schmitz) [inline thirdparty/portmidi/pm_win/pmwinmm.c]: This doesn't seem needed, I could build without it

File: thirdparty/portmidi/pm_win/pmwinmm.c [hunk 3]
```diff
@@ -40,14 +42,14 @@
 
 /* callback routines */
 static void CALLBACK winmm_in_callback(HMIDIIN hMidiIn,
-                                       WORD wMsg, DWORD dwInstance, 
-                                       DWORD dwParam1, DWORD dwParam2);
+										WORD wMsg, DWORD_PTR dwInstance,
+										DWORD_PTR dwParam1, DWORD_PTR dwParam2);
```
Comment (Jojo-Schmitz) [inline thirdparty/portmidi/pm_win/pmwinmm.c]: Here too please avoid the white space changes.

In general we indent by 6 spaces, not by tabs. Not in thirdparty code though, but there we don't change it either

**Comment by Jojo-Schmitz** [comment]
Project seems not as dead as it appears on sourceforge, see https://sourceforge.net/p/portmedia/code/HEAD/tree/portmidi/trunk/pm_win/
which has this fix built in

**Comment by L0gi** [comment]
then it's probably better to pull in the thirdparty code from SF. 

Is there someone maintaining thirdparty libraries, or should I just incorporate them and create a pull request?

**Comment by Jojo-Schmitz** [comment]
Pull and merge with the (hopefully little) changes we did to the code (some of which may be obsolete or could and should be solved differently, e.g. some of the MSVC specific changed to avoid compiler warnings, those should better get disabled in CMakeLists.txt)

**Comment by Jojo-Schmitz** [comment]
See 4028, which does that update to the latest PortMedia code

**Comment by anatoly-os** [comment]
See #4028

**Comment by anatoly-os** [comment]
See #4028

**Comment by anatoly-os** [comment]
See #4028

**Comment by anatoly-os** [comment]
See #4028

**Comment by anatoly-os** [comment]
See #4028

**Pull request closed without merging**